File: minji/week1/10773.py
'''
    [BOJ] 10773 제로
'''
import sys

N = int(sys.stdin.readline().rstrip())
s = []
for _ in range(N):
    num = int(sys.stdin.readline().rstrip())
    if num == 0:
        s.pop()
    else:
        s.append(num)
ans = sum(x for x in s)
print(ans)

File: minji/week1/10773.cpp
/*
	[BOJ] 10773 Á¦·Î
*/
#pragma warning (disable: 4996)

#include <iostream>
#include <vector>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false); cin.tie(0); cout.tie(0);
	vector<int> s;
	int N = 0;
	cin >> N;
	while (N--) {
		int num = 0;
		cin >> num;
		if (num == 0) {
			s.pop_back();
		}
		else {
			s.push_back(num);
		}
	}
	int sum = 0;
	for (int i = 0; i < s.size(); ++i) {
		sum += s[i];
	}
	cout << sum << "\n";
	return 0;
}